src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11652,18 +11662,21 @@ int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexS
             (!_legacy_mode_vl) ||
             (attributes->is_legacy_mode())),"XMM register should be 0-15");
     // Instruction with legacy_mode true should have dst, nds and src < 15
-    assert(((dst_enc < 16 && nds_enc < 16 && src_enc < 16) || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
+    assert(((dst_enc < 16 && nds_enc < 16 && src_enc < 16) || has_gpr || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
   }
 
   clear_managed();
-  if (UseAVX > 2 && !attributes->is_legacy_mode())
+  if (UseAVX > 2 && !attributes->is_legacy_mode() || is_extended)
```

Comment:
No change needed at this line.

Reply:
As in the vex_prefix function case, I left this code as is for now.